app/services/task_category_service.rb

```diff
@@ -0,0 +1,34 @@
+class TaskCategoryService
+  attr_accessor :errors
+
+  def find_all
+    task_categories = TaskCategory.all
+  end
+
+  def create(name, description)
+    task_category = TaskCategory.find_by(name: name)
+    if task_category
+      self.errors = 'Task Category Already Exists with this name'
+    else 
+      TaskCategory.create(name: name, description: description)
```

Comment:
this can return false and we are not checking anywhere if this returned true or false, why?